aten/src/ATen/native/mps/operations/BinaryKernel.mm

```diff
@@ -1,10 +1,387 @@
+  uint32_t otherTensorStorage = otherTensor.storage_offset() * otherTensor.element_size();
+  mps::MPSScalar scalar;
+  if (all_scalar) {
+    type = BinaryKernelType::Scalar;
+    if (iter.is_cpu_scalar(1)) {
+      scalar = mps::getMPSScalar(inputTensor.item(), inputTensor.scalar_type());
+      inputBuffer = (id<MTLBuffer>)getIMPSAllocator()->allocScalarBufferWithValue(&scalar.value, scalar.size).get();
+      inputTensorStorage = 0;
+    }
+    if (iter.is_cpu_scalar(2)) {
+      scalar = mps::getMPSScalar(otherTensor.item(), otherTensor.scalar_type());
+      otherBuffer = (id<MTLBuffer>)getIMPSAllocator()->allocScalarBufferWithValue(&scalar.value, scalar.size).get();
+      otherTensorStorage = 0;
+    }
+  } else if (scalar_pos) {
+    type = allContiguous ? BinaryKernelType::RHS_Scalar : BinaryKernelType::Strided_RHS_Scalar;
+    if (iter.is_cpu_scalar(scalar_pos)) {
+      scalar = mps::getMPSScalar(otherTensor.item(), otherTensor.scalar_type());
+      otherBuffer = (id<MTLBuffer>)getIMPSAllocator()->allocScalarBufferWithValue(&scalar.value, scalar.size).get();
+      otherTensorStorage = 0;
+    }
+  } else {
+    type = allContiguous ? BinaryKernelType::Tensor : BinaryKernelType::Strided_Tensor;
+  }
+
+  const uint32_t nDim = iter.ndim();
+  constexpr uint32_t nOffsets = 3;
+
+  dispatch_sync(mpsStream->queue(), ^(){
+    @autoreleasepool {
+      uint32_t numThreads = iter.numel();
+      id<MTLComputeCommandEncoder> computeEncoder = mpsStream->commandEncoder();
+      MTLSize gridSize = MTLSizeMake(numThreads, 1, 1);
+      const IntArrayRef& iterShape = iter.shape();
+      std::vector<uint32_t> iterShapeData(iterShape.size());
+      std::vector<std::array<uint32_t, nOffsets>> strides(nDim);
+
+      if (!allContiguous) {
+        for (const auto i: c10::irange(iterShape.size())) {
+          TORCH_CHECK(i <= UINT32_MAX);
+          iterShapeData[i] = (uint32_t)(iterShape[i]);
+        }
+
+        for (const auto i: c10::irange(nDim)) {
+          for (const auto offset: c10::irange(nOffsets)) {
+              auto final_offset = offset;
+              if (tensor_pos) {
+                if (offset == 1) {
+                  final_offset = tensor_pos;
+                } else if (offset == 2) {
+                  final_offset = scalar_pos;
+                }
+              }
+              strides[i][offset] = iter.strides(final_offset)[i];
+          }
+        }
+      }
+
+      std::string kernel = op;
+      kernel += "_kernel";
+      if (all_scalar) {
+        kernel += "_scalar";
+      }
+      if (scalar_pos) {
+        kernel += "_scalar_rhs";
+      }
+      if (!allContiguous) {
+        kernel += "_strided";
+      }
+
+      if (allContiguous) {
+        // Round up to next multiple of 4
+        numThreads = (numThreads + (4 - numThreads % 4)) / 4;
+        gridSize = MTLSizeMake(numThreads, 1, 1);
+      }
+
+      id<MTLComputePipelineState> binaryPSO = mps::getBinaryPSO(device,
```

Comment:
If we want this kernel to appear in Profiler, you need to include `MPSProfiler.h` header and add this after getting your PSO (line 308):
`getMPSProfiler().beginProfileKernel(binaryPSO, kernel, {inputTensor, otherTensor});`

Reply:
Thanks Ramin - yes, I'd like these to appear in the profiler. I'll update this